app/services/AuthService.ts

```diff
@@ -31,6 +31,13 @@ export const AuthService = {
     if (PAYLOAD === null) {
       throw new JwtMalformedException();
     }
+
+    // IAT exists but ts kept giving error
+    // @ts-ignore
+    if (PAYLOAD.iat > Date.now()) {
+      throw new JwtMalformedException();
+    }
```

Comment:
Try to avoid using @ts-ignore.

Instead, define the type of the object.

In this case, instead of having the type inference here

```typescript
    const PAYLOAD: JSON = jwt.decode(token, { json: true });
```

You could do something like 

```typescript
const PAYLOAD: { iat: number } = jwt.decode(token, { json: true });
```

Except you'd probably define the type somewhere else like such

```typescript
type JWTData = {
  iat: number
}
```

and just add the rest of the typing stuff.

Then you can use this custom type for both the datatype of PAYLOAD, but also the function return type.